src/d3a/models/strategy/infinite_bus.py

```diff
@@ -0,0 +1,75 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import sys
+
+from d3a.models.strategy.commercial_producer import CommercialStrategy
+from d3a.models.strategy import BidEnabledStrategy
+from d3a.d3a_core.exceptions import MarketException
+from d3a.models.const import ConstSettings
+from d3a.models.read_user_profile import read_arbitrary_profile, InputProfileTypes
+
+INF_ENERGY = int(sys.maxsize)
+
+
+class InfiniteBusStrategy(CommercialStrategy, BidEnabledStrategy):
+    parameters = ('energy_sell_rate', 'energy_buy_rate')
+
+    def __init__(self, energy_sell_rate=None, energy_buy_rate=None):
+        super().__init__()
+        self.energy_per_slot_kWh = INF_ENERGY
+        energy_buy_rate = ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE \
+            if energy_buy_rate is None else energy_buy_rate
+        energy_sell_rate = ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE \
+            if energy_sell_rate is None else energy_sell_rate
+        self.energy_rate = read_arbitrary_profile(InputProfileTypes.IDENTITY, energy_sell_rate)
+        self.energy_buy_rate = read_arbitrary_profile(InputProfileTypes.IDENTITY, energy_buy_rate)
+
+    def offer_energy(self, market):
+        energy_rate = self.energy_rate[market.time_slot]
```

Comment:
It is a bit of an overkill, but it might improve code reuse, therefore I won't hesitate to put it onto the table: instead of overriding the offer_energy CommercialProducer method with a slightly modified version of it here, how about refactoring the CommercialProducer class to parse the self.energy_rate as a profile (like you already do it in the constructor of the InfiniteBusStrategy). That way the offer_energy method can be deleted, and as an added bonus, parts of the InfiniteBusStrategy constructor will be handled automatically by the base class constructor (CommercialStrategy)